site/index.html

```diff
@@ -67,35 +70,103 @@
 
               content = content
                 .replace(titleBlock, titleHeading)
-                .replace(asciinemaDirective, (_, fileName) => `<asciinema-player src="${ window.location.origin}/static/casts/${fileName}.cast"></asciinema-player>`)
+                .replace(
+                  asciinemaDirective,
+                  (_, fileName) =>
+                    `<asciinema-player src="${window.location.origin}/static/casts/${fileName}.cast"></asciinema-player>`
+                )
                 .replace(hugoHideDirectives, "")
                 .replace(hugoDirectiveTags, "");
               return content;
             });
 
-            hook.doneEach(function() { 
-              const tocLists = Array.from(document.getElementsByClassName("sidebar-nav").item(0)?.getElementsByTagName("ul"))
-              
+            hook.beforeEach(function (content) {

```

Comment:
extract out functions to make this code more readable. Add comments as appropriate to explain why we need this function.

Reply:
Separated and commented.